bleak/backends/bumble/client.py

```diff
@@ -0,0 +1,293 @@
+from ..service import BleakGATTServiceCollection
+
+logger = logging.getLogger(__name__)
+CLIENT_MAC = "F0:F1:F2:F3:F4:00"
+
+
+class BleakClientBumble(BaseBleakClient, Device.Listener, Connection.Listener):
+    """Bumble class interface for BleakClient
+
+    Args:
+        address_or_ble_device (`BLEDevice` or str): The Bluetooth address of the BLE peripheral to connect to or the `BLEDevice` object representing it.
+
+    Keyword Args:
+        adapter (str): Bumble transport adapter to use. Defaults to `tcp-server:_:1234`.
+
+    """
+
+    def __init__(
+        self,
+        address_or_ble_device: Union[BLEDevice, str],
+        **kwargs,
+    ):
+        super(BleakClientBumble, self).__init__(address_or_ble_device, **kwargs)
+        self._dev = Device("client", address=Address(CLIENT_MAC))
+        self._is_connected = False
+        self._connection = None
+        self._peer = None
+        self._dev.host = Host()
+        self._dev.host.controller = Controller("Client", link=get_link())
+        self._dev.listener = self
+        self._adapter: Optional[Transport] = kwargs.get(
+            "adapter", get_default_transport()
+        )
+
+    @property
+    def mtu_size(self) -> int:
+        return self._connection.att_mtu
+
+    async def connect(self, **kwargs) -> bool:
+        """Connect to the specified GATT server.
+
+        Returns:
+            Boolean representing connection status.
+
+        """
+        await start_transport(self._adapter)
+        await self._dev.power_on()
+        await self._dev.connect(self.address)
+
+        self.services = await self.get_services()
+
+    async def disconnect(self) -> bool:
+        """Disconnect from the specified GATT server.
+
+        Returns:
+            Boolean representing connection status.
+
+        """
+        await self._dev.disconnect(
+            self._connection, HCI_REMOTE_USER_TERMINATED_CONNECTION_ERROR
+        )
+
+    async def pair(self, *args, **kwargs) -> bool:
+        """Pair with the peripheral."""
+        await self._peer.connection.pair()
+
+    async def unpair(self) -> bool:
+        """Unpair with the peripheral."""
+        raise NotImplementedError()
+
+    @property
+    def is_connected(self) -> bool:
+        """Check connection status between this client and the server.
+
+        Returns:
+            Boolean representing connection status.
+
+        """
+        return self._is_connected
+
+    async def get_services(self, **kwargs) -> BleakGATTServiceCollection:
+        """Get all services registered for this GATT server.
+
+        Returns:
+           A :py:class:`bleak.backends.service.BleakGATTServiceCollection` with this device's services tree.
+
+        """
+        if not self.is_connected:
+            raise BleakError("Not connected")
+
+        if self.services is not None:
+            return self.services
+
+        new_services = BleakGATTServiceCollection()
+        self._peer = Peer(self._connection)
```

Comment:
This is a type error; self._peer already has static type None - is it something OR nothing or is it always guaranteed to be something?  Should it be a mutable variable that is global to the class or should it be an immutable variable local to the function it's used in?